index.js

```diff
@@ -1,43 +1,73 @@
+// zb2rhiVd5G2DSpnbYtty8NhYHeDvNkPxjSqA7YbDPuhdihj9L
+function decodeCID (value) {
+  const cid = new CID(value).toJSON()
+  if (cid.version === 0) {
+    return decodeCidV0(value, cid)
+  }
+  if (cid.version === 1) {
+    return decodeCidV1(value, cid)
+  }
+  throw new Error('Unknown CID version', cid.version, cid)
+}
+
+function decodeCidV0 (value, cid) {
+  return {
+    cid,
+    multibase: {
+      name: 'base58btc',
+      code: 'implicit'
+    },
+    multicodec: {
+      name: cid.codec,
+      code: 'implicit'
+    },
+    multihash: multihash.decode(cid.hash)
+  }
+}
+
+function decodeCidV1 (value, cid) {
+  return {
+    cid,
+    multibase: multibaseConstants.codes[value.substring(0, 1)],
+    multicodec: {
+      name: cid.codec,
+      code: '0x' + mutlicodecVarintTable[cid.codec].toString('hex')
+    },
+    multihash: multihash.decode(cid.hash)
+  }
 }
 
 document.addEventListener('DOMContentLoaded', () => {
   const output = document.querySelector('#cid')
   const input = document.querySelector('#input-cid')
   const multihashOutput = document.querySelector('#multihash')
+  const multicodecOutput = document.querySelector('#multicodec')
   const multibaseOutput = document.querySelector('#multibase')
+  const humanReadableCidOutput = document.querySelector('#hr-cid')
 
   function setOutput (output, value) {
     window.location.hash = value
     try {
-      const cid = decodeCID(value)
-      // cidv0 ::= <multihash-content-address>
-      // QmRds34t1KFiatDY6yJFj8U9VPTLvSMsR63y7qdUV3RMmT
-      // <cidv1> ::= <multibase-prefix><cid-version><multicodec-packed-content-type><multihash-content-address>
-      // zb2rhiVd5G2DSpnbYtty8NhYHeDvNkPxjSqA7YbDPuhdihj9L
-      if (cid.version === 0) {
-        multibaseOutput.innerText = `Multibase: base58btc`
-      }
-      if (cid.version === 1) {
-        const code = value.substring(0, 1)
-        const multibaseCode = multibases.codes[code]
-        multibaseOutput.innerText = `Multibase: ${multibaseCode.name}`
-      }
-      const mh = multihash.decode(cid.hash)
-      output.innerText = `Codec: ${cid.codec} Version: ${cid.version}`
-      multihashOutput.innerText = `Code: ${mh.code} Hash Function: ${mh.name} Length: ${mh.length}`
+      const data = decodeCID(value.trim())
+      console.log(data)
+      const hrCid = `${data.multibase.name} - cidv${data.cid.version} - ${data.cid.codec} - ${data.multihash.name}-${data.multihash.length * 8}-${data.multihash.digest.toString('hex')}`
+      humanReadableCidOutput.innerText = hrCid
+      multibaseOutput.innerHTML = toDefinitionList({code: data.multibase.code, name: data.multibase.name})
+      multicodecOutput.innerHTML = toDefinitionList({code: data.multicodec.code, name: data.multicodec.name})
+      multihashOutput.innerHTML = toDefinitionList({code: data.multihash.code, name: data.multihash.name, bits: data.multihash.length * 8})
     } catch (err) {
-      output.innerText = err
+      console.log(err)
```

Comment:
Hm, we should still show this somehow, as it's probably a user error at this point and not program error. Passing in a invalid multihash should show the error somehow.